client/my-sites/site-settings/settings-p2/preapproved-domains.jsx

```diff
@@ -0,0 +1,277 @@
+import { Card } from '@automattic/components';
+import { ToggleControl } from '@wordpress/components';
+import classNames from 'classnames';
+import debugModule from 'debug';
+import { flowRight, includes, pickBy, filter } from 'lodash';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import QuerySiteSettings from 'calypso/components/data/query-site-settings';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormSettingExplanation from 'calypso/components/forms/form-setting-explanation';
+import TokenField from 'calypso/components/token-field';
+import wpcom from 'calypso/lib/wp';
+import SettingsSectionHeader from 'calypso/my-sites/site-settings/settings-section-header';
+import isSiteP2Hub from 'calypso/state/selectors/is-site-p2-hub';
+import isSiteWPForTeams from 'calypso/state/selectors/is-site-wpforteams';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import wrapSettingsForm from '../wrap-settings-form';
+
+const debug = debugModule( 'calypso:my-sites:settings:p2-settings' );
+export class P2PreapprovedDomainsForm extends Component {
```

Comment:
Would we use a function component here instead of a class? The hooks could maybe make the state changes easier to do.

Reply:
🤔 Good question @naxoc . I think I kind of need `getDerivedStateFromProps` to manage the toggle button which has state (can be manually toggled on and off), but will have to be force-toggled on when `props.fields.p2_preapproved_domains` is not empty (on initial settings load, for example). Would that be doable, or maybe easier even, with a function component?

Reply:
Done and thanks for the suggestion, @naxoc.